modules/videoio/src/container_avi.cpp

```diff
@@ -3,10 +3,29 @@
 // of this distribution and at http://opencv.org/license.html.
 
 #include "opencv2/videoio/container_avi.private.hpp"
+#include <fstream>
+#include <limits>
+#include <typeinfo>
 
 namespace cv
 {
 
+// Utility function for safe integer conversions
+template <typename D, typename S>
+inline D safe_int_cast(S val)
+{
+    typedef std::numeric_limits<S> st;
+    typedef std::numeric_limits<D> dt;
+    CV_StaticAssert(st::is_integer && dt::is_integer, "Integer type is expected");
+    const bool in_range_r = (double)val <= (double)dt::max();
+    const bool in_range_l = (double)val >= (double)dt::min();
+    if (!in_range_r || !in_range_l)
```

Comment:
since we know the exact context, can we have more specific error message, e.g. "Too long AVI video; it can hold no more than 4Gb of data. Consider another type of container"

Reply:
BTW, It would be nice to have similar function for other part of code too, at least `imgcodecs`.